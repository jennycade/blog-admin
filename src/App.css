body {
  margin: 0;
  font-family: sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  min-height: 100vh;
  background: linear-gradient(163deg, #C6FFDD, #FBD786, #f7797d);
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

main {
  position: relative;
  margin: 0 1rem 1rem 1rem;
  background-color: white;
  padding: 1rem;
  border-radius: 5px;
}

/* POSITION */
.top-right {
  position: absolute;
  top: calc(1rem + 21.440px);
  right: 1rem;
}

/* LINKS */
a {
  color: #1456b8;
}
a:visited {
  color: #583d8f;
}

/* BUTTONS */
button {
  border: none;
  padding: 0.5rem;
  border-radius: 3px;
  font-size: medium;
  font-weight: bold;
  background-color: #0a5c2b;
  border: 1px solid #0a5c2b;
  color: white;
  transition: filter 200ms ease-in-out;
}
button:hover, .link-button:hover {
  filter: brightness(120%);
}
/* fake button */
.link-button {
  display: block;
  border: none;
  padding: 0.5rem;
  border-radius: 3px;
  font-size: medium;
  font-weight: bold;
  background-color: #0a5c2b;
  border: 1px solid #0a5c2b;
  color: white;
  transition: filter 200ms ease-in-out;
  cursor: default;
}
a.link-button, a.link-button:visited {
  text-decoration: none;
  color: white;
}

.btn-primary {
  background-color: #0a5c2b;
  border-color: #0a5c2b;
  color: white;
}
.btn-warning {
  background-color: white;
  color: #f7797d;
  border-color: #f7797d;
}
.btn-link {
  background-color: transparent;
  border: none;
  margin: 0;
  padding: 0;
  color: #1456b8;
  font-weight: inherit;
}
.button-group {
  display: flex;
  margin: 5px 0;
  gap: 5px;
}

/* DROPDOWNS */
.dropdown-toggle {
  /* vertically align text with dropdown icon */
  display: flex;
  align-items: center;
}
.dropdown {
  position: absolute;
  margin: 0;
  padding: 0;
  z-index: 2;

  text-align: right;
  list-style: none;

  display: flex;
  flex-direction: column;

  background-color: white;
  box-shadow: 2px 3px 10px 1px rgba(0, 0, 0, 0.05);
  border-radius: 0 0 3px 3px;
}
.dropdown li {
  padding: 0.5rem;
}

/* ALERTS */
.alert-container {
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  backdrop-filter: brightness(50%);
}
.alert {
  /* TODO: improve on mobile (too narrow) */
  position: fixed;
  top: 50%;
  left: 50%;
  /* bring your own prefixes */
  transform: translate(-50%, -50%);
  max-width: 600px;
  border-radius: 5px;
  padding: 2rem;

  background-color: white;
}
.alert .button-group {
  justify-content: center;
}

/* FORMS */
.form-field {
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
}
.form-field label {
  font-weight: bold;
}
.form-field textarea {
  height: 10rem;
}

/* TABLES */
.table-container {
  width: 100%;
  overflow-x: auto;
}
table {
  margin: 0 auto;
  border-collapse: collapse;
  min-width: min-content;
}
tr {
  border-bottom: 1px solid #afb6b2;
}
td, th {
  padding: 1rem 0.5rem;
}

/* DEFINITION LISTS */
dt {
  margin-top: 1rem;
  font-weight: bold;
}
dd {
  margin: 0;
  padding: 0;
}

/* UNORDERED LISTS */
ul {
  margin: 0;
  padding: 0;
  list-style: none;
}
ul>li {
  margin-bottom: 1rem;
}

/* COLLAPSIBLE */
.collapsible.collapsed:not(:focus):not(:active) {
  clip: rect(0 0 0 0); 
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap; 
  width: 1px;
}

/* desktops */
@media (min-width: 600px) {
  .collapsible.collapse-mobile.collapsed:not(:focus):not(:active) {
    clip: inherit;
    clip-path: inherit;
    height: inherit;
    overflow: inherit;
    position: inherit;
    white-space: inherit;
    width: inherit;
  }
}