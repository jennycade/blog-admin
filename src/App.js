// import './App.css';
import 'bootstrap/dist/css/bootstrap.css';

import React from 'react';
import { useState, useEffect } from 'react';

import {
  Routes, Route, 
} from 'react-router-dom';

// routes
import Home from './routes/Home';
import Posts from './routes/Posts';
import Users from './routes/Users';
import Comments from './routes/Comments';

// components
import SignInForm from './SignInForm';
import NavBar from './NavBar';
import NavBarExample from './examples/navbarExample';

// API stuff
import {
  shouldUseLocalStorageTokenAndUser,
  getLocalStorageTokenAndUser,
  signInUser
} from './databaseFunctions';

const App = () => {
  // state
  const [token, setToken] = useState('');
  const [currentUser, setCurrentUser] = useState({});

  // effects
  // check for localStorage token
  useEffect(() => {
    if (!token) {
      if (shouldUseLocalStorageTokenAndUser()) {
        const results = getLocalStorageTokenAndUser();
        setToken(results.token);
        setCurrentUser(results.user);
      }
    }
  }, [token]);

  // functions
  const handleSignInSubmit = async (username, password) => {
    // try to sign in
    const result = await signInUser(username, password);
    if (result.errorMessage) {
      alert(result.errorMessage);
    } else {
      setToken(result.token);
      setCurrentUser(result.user);
    }
  }

  return (
    <div className='app col-lg-8 mx-auto p-3 py-md-5'>
      {/* SIGN IN FORM */}
      { token === '' && (
        <SignInForm handleSignInSubmit={handleSignInSubmit} />
      )}

      {/* WELCOME */}
      { token !== '' && (
        <>
          <NavBar />

          <main>
            <Routes>
              <Route path='/' element={
                <Home />
              } />
              <Route path='posts' element={
                <Posts />
              } />
              <Route path='users' element={
                <Users />
              } />
              <Route path='comments' element={
                <Comments />
              } />
            </Routes>

          </main>
        </>
      )}
    </div>
  );
}

export default App;
