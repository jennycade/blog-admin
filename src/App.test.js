import React from 'react';
import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('App component', () => {
  // it('matches previous snapshot', () => {
  //   const { container } = render(<App />);
  //   expect(container).toMatchSnapshot();
  // });

  // it('renders radical rhinos after button click', () => {
  //   // get button to click
  //   render(<App />);
  //   const button = screen.getByRole('button', { name: 'Click Me'});

  //   // click it
  //   userEvent.click(button);

  //   // changes heading
  //   expect(screen.getByRole('heading').textContent).toMatch(/radical rhinos/i);
  // });

  it('shows sign in form on load', () => {
    render(<App />);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });
});