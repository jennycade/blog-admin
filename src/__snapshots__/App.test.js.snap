// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches previous snapshot 1`] = `
<div>
  <div
    class="app"
  >
    <form>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="usernameInput"
        >
          Username
        </label>
        <input
          class="form-control"
          id="usernameInput"
          type="text"
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="passwordInput"
        >
          Password
        </label>
        <input
          class="form-control"
          id="passwordInput"
          type="password"
        />
      </div>
      <div
        class="mb-3"
      >
        <button
          class="btn btn-primary"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
</div>
`;
